// cisco-ui-react/src/components/Button.jsx

import React  from 'react';
import PropTypes from 'prop-types';

import CUI_CONST from './CiscoUIConstants.jsx';

import Icon from './Icon.jsx';

class Button extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            disabled: props.disabled,
        };
    }

    render() {
        let classes = ['btn', `btn--${this.props.size}`];

        if(this.props.color){
            classes.push(`btn--${this.props.color}`);
        }
        if(this.props.icon){
            classes.push(`btn--icon`);
        }

        return (
            <button
                className={classes.join(' ')}
                disabled={this.state.disabled}

                {...this.props}
            >
                {this.props.icon &&
                    <Icon icon={this.props.icon} />
                }
                {this.props.label}
                {this.props.children}
            </button>
        );
    }
}

Button.defaultProps = {
    color: 'default',
    label: null,
    disabled: false,
    icon: null,
    size: 'default',
};


Button.propTypes = {
    color: PropTypes.oneOf(CUI_CONST.BUTTON_COLORS),
    label: PropTypes.string,
    disabled: PropTypes.bool,
    icon: PropTypes.string,
    size: PropTypes.oneOf(CUI_CONST.BUTTON_SIZES),
};

export default Button;
