// cisco-ui-react/src/components/Checkbox.jsx
import React  from 'react';
import PropTypes from 'prop-types';

import CUI_CONST from './CiscoUIConstants.jsx';

// import _ from 'lodash';

/**
 * Checkbox
 * allowed props: label, id, name, required
 * allowed state: value, disabled, checked
 */
class Checkbox extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            disabled: props.disabled,
            checked: props.checked,
            required: props.required,
            value: props.value,
        };

        this.onChange = this.onChange.bind(this);
    }

    onChange(event) {
        return this.setState({checked: event.target.checked});
    }

    render() {
        return (
            <label className="checkbox" disabled={this.state.disabled}>
                <input
                    type="checkbox"
                    id={this.props.id}
                    name={this.props.name || this.props.id}

                    checked={this.state.checked}
                    required={this.state.required}
                    value={this.state.value}
                    disabled={this.state.disabled}

                    onChange={this.onChange}
                    
                    {...this.props}
                />

                <span className="checkbox__input" />

                {this.props.label &&
                    <span className="checkbox__label">{this.props.label}</span>
                }
            </label>
        )
    }
}

Checkbox.defaultProps = {
    disabled: false,
    checked: false,
    required: false,
};


Checkbox.propTypes = {
    id: PropTypes.string,
    name: PropTypes.string,
    disabled: PropTypes.bool,
    checked: PropTypes.bool,
    required: PropTypes.bool,
    value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
};

export default Checkbox;
