// cisco-ui-react/src/components/Footer.jsx

import React  from 'react';
import Icon from './Icon.jsx';
import PropTypes from 'prop-types';

import CUI_CONST from './CiscoUIConstants.jsx';

class Footer extends React.Component {
    render() {


        let classes = `footer footer--new footer--${this.props.padding}`;

        if(this.props.basic) {
            classes += ' footer--basic';
        }

        return (
            <footer className={classes}>

                <div className="footer__links">
                    {this.props.children &&
                        <ul className="list">
                            {this.props.children}
                        </ul>
                    }
                </div>

                {!this.props.basic &&

                    <div className="footer__legal">
                        <div className="footer__logo">
                            <a href="http://www.cisco.com" target="_blank">
                                <span className="icon-cisco" />
                            </a>
                        </div>
                        <div className="footer__copyright">
                            <div>Copyright &copy; {new Date().getFullYear()} Cisco Systems Inc.</div>
                            <div>All rights reserved.</div>
                        </div>
                    </div>

                }
            </footer>
        )
    }
}

Footer.defaultProps = {
    links: [],
    basic: false,
    padding: 'default',
};

Footer.propTypes = {
    basic: PropTypes.bool,
    padding: PropTypes.oneOf(CUI_CONST.FOOTER_PADDING),
    links: PropTypes.arrayOf(
        PropTypes.oneOfType([
            PropTypes.string,
            PropTypes.shape({
                href: PropTypes.string,
                label: PropTypes.oneOfType([
                    PropTypes.number,
                    PropTypes.string,
                ]),
            })
        ])
    ),
};

export default Footer;
