// cisco-ui-react/src/components/Hidden.jsx

import React  from 'react';
import PropTypes from 'prop-types';

import CUI_CONST from './CiscoUIConstants.jsx';

class Hidden extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            value: props.value,
            required: props.required,
        };

        this.onChange = this.onChange.bind(this);
    }

    onChange(event){
        return this.setState({value: event.target.value});
    }

    render() {
        return (
            <input type="hidden"
                id={this.props.id}
                name={this.props.name && this.props.id}

                required={this.state.required}
                value={this.state.value}
            />
        )
    }
}

Hidden.defaultProps = {
    value: '',
    disabled: false,
    required: false,
};

Hidden.propTypes = {
    id: PropTypes.string,
    name: PropTypes.string,
    disabled: PropTypes.bool,
    required: PropTypes.bool,
    value: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
    ]),
};

export default Hidden;
