// cisco-ui-react/src/components/Radio.jsx

import React  from 'react';
import PropTypes from 'prop-types';
import CUI_CONST from './CiscoUIConstants.jsx';

import _ from 'lodash';

/**
 * Radio
 * allowed props: label, id, name, required
 * allowed state: value, disabled, checked
 */
class Radio extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        let dataProps = _.pickBy(this.props, function(val, key){
            return /^data/i.test(key);
        });

        return (
            <label
                className={`radio radio--${this.props.layout}`}
                disabled={this.props.disabled}

                {...dataProps}
            >
                <input
                    type="radio"
                    id={this.props.id}
                    name={this.props.name || this.props.id}

                    checked={this.props.checked}
                    required={this.props.required}
                    value={this.props.value}
                    disabled={this.props.disabled}
                    onChange={this.props.onChange}
                />

                <span className="radio__input" />

                {this.props.label &&
                    <span className="radio__label">{this.props.label}</span>
                }
            </label>
        );


    }
}

Radio.defaultProps = {
    disabled: false,
    checked: false,
    required: false,
    value: '',
    layout: 'inline',
    onChange: function(){},
};


Radio.propTypes = {
    id: PropTypes.string,
    name: PropTypes.string,
    disabled: PropTypes.bool,
    checked: PropTypes.bool,
    required: PropTypes.bool,
    value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
    layout: PropTypes.oneOf(CUI_CONST.RADIO_LAYOUTS),
    onChange: PropTypes.func,
};

export default Radio;
