// cisco-ui-react/src/components/Scoreboard.jsx

import React  from 'react';
import PropTypes from 'prop-types';
import CUI_CONST from './CiscoUIConstants.jsx';

class Scoreboard extends React.Component {
    render() {
        let classes = `scoreboard scoreboard--${this.props.size} scoreboard--${this.props.padding}`;

        if(this.props.raised) {
            classes += ' scoreboard--raised';
        }
        if(this.props.inline) {
            classes += ' scoreboard--inline';
        }

        return (
            <div className={classes}>
                <div className="scoreboard__title">{this.props.title}</div>
                {this.props.divided &&
                    <hr className="text-muted" />
                }
                <div className="scoreboard__subtitle">{this.props.subtitle}</div>
            </div>
        );
    }
}

Scoreboard.defaultProps = {
    size: 'default',
    padding: 'regular',
    divided: false,
    raised: false,
    inline: false,
};


Scoreboard.propTypes = {
    size: PropTypes.oneOf(CUI_CONST.SCOREBOARD_SIZES),
    padding: PropTypes.oneOf(CUI_CONST.SCOREBOARD_PADDING),
    title: PropTypes.string,
    subtitle: PropTypes.string,
    divided: PropTypes.bool,
    raised: PropTypes.bool,
    inline: PropTypes.bool,
};

export default Scoreboard;
