// cisco-ui-react/src/components/Sidebar.jsx
import React  from 'react';
import PropTypes from 'prop-types';
import CUI_CONST from './CiscoUIConstants.jsx';

class Sidebar extends React.Component {
    render() {
        return (
            <div className="sidebar">
                <div className="sidebar__header">
                    <div className="sidebar__header-title">{this.props.title}</div>
                </div>
                <ul>
                    {this.props.navigation.map(function(nav){
                        return (
                            <li className="sidebar__item" key={nav.name + nav.link}>
                                <a href={nav.link}>
                                    {nav.name}
                                </a>
                            </li>
                        )
                    })}
                </ul>
            </div>
        )
    }
}

Sidebar.propTypes = {
    title: PropTypes.string,
    navigation: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        link: PropTypes.string,
    })),
};

Sidebar.defaultProps = {
    title: "Title",
    navigation: []
};

export default Sidebar;
