// cisco-ui-react/src/components/Tabs.jsx

import React from 'react';
import PropTypes from 'prop-types';

import CUI_CONST from './CiscoUIConstants.jsx';

class Tab extends React.Component{

    constructor(props){
        super(props);

        this.onClick = this.onClick.bind(this);
    }

    onClick(){
        return this.props.onChange(this.props.value);
    }

    render(){

        let classes = ['tab'];
            if(this.props.active){
                classes.push('active');
            }

            return (
                <li className={classes.join(' ')}>
                    <a onClick={this.onClick}>
                        <div className="tab__heading" title={this.props.label}>{this.props.label}</div>
                    </a>
                </li>
            )

    }
}

Tab.defaultProps = {
    value: '',
    label: '',
    active: false,
    onClick: function(){},
    title: null,
};

Tab.propTypes = {
    value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
    label: PropTypes.node,
    active: PropTypes.bool,
    title: PropTypes.string,
};



class Tabs extends React.Component {

    constructor(props){
        super(props);
    }

    render() {

        let items = this.props.items.map((item) => {
            return (
                <Tab
                    key={item.value}

                    value={item.value}
                    label={item.label}
                    title={item.title}

                    active={this.props.selected === item.value}
                    onChange={this.props.onChange}
                />
            );
        });

        return (
            <div className="secondary-tabs">
                <ul className="tabs">
                    {items}
                </ul>
            </div>
        );
    }
}

Tabs.defaultProps = {
    items: [],
    selected: '',
    onChange: function(){},
};

Tabs.propTypes = {
    items: PropTypes.arrayOf(PropTypes.object),
    selected: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string,
    ]),
    onChange: PropTypes.func,
};

export default Tabs;
