// cisco-ui-react/src/components/Textarea.jsx

import React  from 'react';
import PropTypes from 'prop-types';
import CUI_CONST from './CiscoUIConstants.jsx';

class Textarea extends React.Component {

    constructor(props) {
        super(props);

        // TODO: remove state from components. See Issue #18
        this.state = {
            value: props.value,
            required: props.required,
        };

        this.onChange = this.onChange.bind(this);
    }

    onChange(event){
        if(this.props.onChange) {
            this.props.onChange(event.target.value);
        }

        return this.setState({value: event.target.value});
    }

    render() {
        return (
            <div className="form-group__text" disabled={this.state.disabled}>
                <textarea
                    id={this.props.id}
                    name={this.props.name || this.props.id}

                    required={this.state.required}
                    disabled={this.state.disabled}

                    onChange={this.onChange}

                    defaultValue={this.state.value}
                    rows={this.props.rows}
                    ref={(input) => { this.input = input; }}
                />

                {this.props.label &&
                    <label htmlFor={this.props.id}>
                        {this.props.label}
                    </label>
                }
            </div>
        )
    }
}

Textarea.defaultProps = {
    value: '',
    required: false,
    disabled: false,
    rows: 1,
    onChange: function(){}
};


Textarea.propTypes = {
    id: PropTypes.string,
    name: PropTypes.string,
    disabled: PropTypes.bool,
    required: PropTypes.bool,
    rows: PropTypes.number,
    onChange: PropTypes.func,
    value: PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
    ]),
};

export default Textarea;
